### PR TITLE
ddt removal in location and permission tests

### PR DESCRIPTION
**Pull request by sthirugn**
This PR also 
* updated filter() usage with list comprehensions and made pylint 100%
* Fixed a jenkins test failure since the permissions are updated in satellite.

### CODE REVIEW

**Comment by sthirugn** [comment]
test_location results:
```sh
----------------------------------------------------------------------
Ran 27 tests in 121.543s

OK
```


test_permission results:
```sh
----------------------------------------------------------------------
Ran 7 tests in 96.693s

OK
```

File: tests/foreman/api/test_location.py [hunk 1]
```diff
@@ -35,10 +26,12 @@ def test_create_location_with_different_names(self, name):
...
         """
-        location = entities.Location(name=name).create()
-        self.assertEqual(location.name, name)
+        for name in valid_data_list():
+            with self.subTest(name):
+                location = entities.Location(name=name).create()
+                self.assertEqual(location.name, name)
 
-    def test_create_location_with_description(self):
+    def test_create_loc_with_desc(self):
```
Comment (elyezer) [inline tests/foreman/api/test_location.py]: I prefer long names and silent pylint, than start shortening them.

File: tests/foreman/api/test_location.py [hunk 2]
```diff
@@ -110,11 +103,11 @@ def test_create_location_with_template(self):
         location = entities.Location(config_template=[template]).create()
         self.assertGreaterEqual(len(location.config_template), 1)
         self.assertNotEqual(
-            filter(lambda ct: ct.id == template.id, location.config_template),
+            [ct for ct in location.config_template if ct.id == template.id],
```
Comment (elyezer) [inline tests/foreman/api/test_location.py]: :+1:

**Comment by elyezer** [comment]
ACK, pending method rename comment.

File: tests/foreman/api/test_location.py (hunk 1 above)
Comment (sthirugn) [inline tests/foreman/api/test_location.py]: I like the function names or test names to be short and crisp.  Anything more can be added in the docstring as necessary.

And IMHO all the shortforms used here - loc, env, cr, desc - are all familiar ones and it should not affect readability :)

**Comment by sthirugn** [comment]
rebased

**Comment by omaciel** [comment]
ACK

**Comment by sthirugn** [comment]
Part of https://github.com/SatelliteQE/robottelo/issues/2769

**Pull request merged**